### PR TITLE
[WEB-2727] - dialog for demo-pro upgrade

### PR DESCRIPTION
**Pull request by RavenHursT**
Also corrects use of dynamic Tailwind classes in the alert banner per: https://stackoverflow.com/a/74959709/282250

### CODE REVIEW

File: app/(candidate)/dashboard/upgrade-to-pro/components/UpdateToProPage.js [hunk 1]
```diff
@@ -52,11 +79,24 @@ export default function DetailsPage(props) {
             is reinvested into making it possible for more people across the
             country to run, win and serve.
           </Body2>
-          <Link href="/dashboard/pro-sign-up">
-            <PrimaryButton className="flex items-center justify-center mx-auto mb-16 w-full md:w-auto">
+          <Link onClick={handleLinkOnClick} href="/dashboard/pro-sign-up">
```
Comment (tomer-tgp) [inline app/(candidate)/dashboard/upgrade-to-pro/components/UpdateToProPage.js:82]: The user can still see the link when they hover over the link.
Wouldn't it be better to go with
```
<Link href={demoPersona ? '#' : '/dashboard/pro-sign/up'}>
```

Comment (tomer-tgp) [reply app/(candidate)/dashboard/upgrade-to-pro/components/UpdateToProPage.js:82]: Also, if a demo user goes directly to `/dashboard/pro-sign-up` will they be able to upgrade? we need to make sure we block that.

**Review by tomer-tgp — CHANGES_REQUESTED** [review]

File: app/(candidate)/dashboard/upgrade-to-pro/components/UpdateToProPage.js (hunk 1 above)
Comment (RavenHursT) [reply app/(candidate)/dashboard/upgrade-to-pro/components/UpdateToProPage.js]: https://github.com/thegoodparty/gp-webapp/pull/244/commits/cac7da553149567db8c8aabfa2994981cdc039d6

**Comment by RavenHursT** [comment]
Going to restyle `AlertDialog` and make button labels configurable for now..

**Review by tomer-tgp — APPROVED** [review]

**Pull request merged**